### DISCUSSION TITLE
Missing type hints

### DISCUSSION THREAD

**Discussion by maroux** [carltongibson/django-filter#1338] [Ideas]
This library is missing type hints, so tools like mypy can't be run:
```
error: Skipping analyzing 'django_filters': found module but no type hints or library stubs
note: See https://mypy.readthedocs.io/en/latest/running_mypy.html#missing-imports
```

Is there any interest in supporting this? I can look into it if so..

**Comment by carltongibson** [comment]
Hi @maroux. 

I'd happy add **some** hints if they're clean, and we can find a way of making then _hints_, rather than strict types.

What I mean by that latter is that I don't want to see things like `Any` or `str|Any` or even really `Dict[str:Any]` — In the first case I'd want no type hint (we can assume Any) in the second just `str` (YES, we know it's Python) and in the last I'm just a be Meh. 

`mypy` has a billion flags. How can we run it so that it's not too Stalinist about these things? 

If we can answer that question then I'd be happy to consider it. 
Thanks!

**Comment by jerivas** [comment]
I just discovered [django-filters-stubs](https://pypi.org/project/django-filter-stubs/) and simply installing it seems to resolve most mypy problems. It hasn't been updated in a while but django-filter's API is pretty stable as well
